src/Altinn.AccessManagement.Core/Services/SingleRightsService.cs

```diff
@@ -25,32 +27,36 @@ public class SingleRightsService : ISingleRightsService
         /// <param name="contextRetrievalService">Service for retrieving context information</param>
         /// <param name="pip">Service implementation for policy information point</param>
         /// <param name="pap">Service implementation for policy administration point</param>
-        public SingleRightsService(IDelegationMetadataRepository delegationRepository, IContextRetrievalService contextRetrievalService, IPolicyInformationPoint pip, IPolicyAdministrationPoint pap)
+        public SingleRightsService(IDelegationMetadataRepository delegationRepository, IContextRetrievalService contextRetrievalService, IPolicyInformationPoint pip, IPolicyAdministrationPoint pap, IAltinn2RightsClient altinn2RightsClient)
         {
             _delegationRepository = delegationRepository;
             _contextRetrievalService = contextRetrievalService;
             _pip = pip;
             _pap = pap;
+            _altinn2RightsClient = altinn2RightsClient;
         }
 
         /// <inheritdoc/>
-        public async Task<DelegationCheckResult> RightsDelegationCheck(int authenticatedUserId, int authenticatedUserAuthlevel, RightsDelegationCheckRequest request)
+        public async Task<DelegationCheckResponse> RightsDelegationCheck(int authenticatedUserId, int authenticatedUserAuthlevel, RightsDelegationCheckRequest request)
         {
-            (DelegationCheckResult result, ServiceResource resource, Party fromParty) = await ValidateRightsDelegationCheckRequest(request);
+            (DelegationCheckResponse result, ServiceResource resource, Party fromParty) = await ValidateRightDelegationCheckRequest(request);
             if (!result.IsValid)
             {
                 return result;
             }
 
+            RightsQuery rightsQuery;
             DelegationHelper.TryGetResourceFromAttributeMatch(request.Resource, out ResourceAttributeMatchType resourceMatchType, out string resourceRegistryId, out string org, out string app, out string serviceCode, out string serviceEditionCode);
             if (resource.ResourceType == ResourceType.Altinn2Service)
             {
-                result.Errors.Add("right[0].Resource", $"Altinn apps and Altinn 2 services are not yet supported. {resource}"); //// ToDo: Update when support exists
-                return result;
+                return await _altinn2RightsClient.PostDelegationCheck(authenticatedUserId, request);
```

Comment:
In stead of sending the entire request object as input to _altinn2RightsClient and have to parse this again you can just give properties as input individually here where serviceCodes, fromParty.PartyId etc is readily available

Reply:
changed it to take the properties it needs instead of the whole request